test/test_tensorclass_nofuture.py

```diff
@@ -354,6 +354,52 @@ class MyDataNested:
     ):
         data[:2] = data3[:2]
 
+    # regression test PR #203
+    # We should be able to set tensors items with MemmapTensors and viceversa
+    @tensorclass
+    class MyDataMemMap1:
+        x: torch.Tensor
+        y: MemmapTensor
+
+    @tensorclass
+    class MyDataMemMap2:
+        x: MemmapTensor
+        y: torch.Tensor
+
+    data1 = MyDataMemMap1(
+        x=torch.zeros(3, 4, 5),
+        y=MemmapTensor.from_tensor(torch.zeros(3, 4, 5)),
+        batch_size=[3, 4],
+    )
+    data2_wrong = MyDataMemMap2(
+        x=MemmapTensor.from_tensor(torch.ones(3, 4, 5)),
+        y=torch.ones(3, 4, 5),
+        batch_size=[3, 4],
+    )
+    with pytest.raises(
+        ValueError, match="__setitem__ is only allowed for same-class assignement"
+    ):
+        data1[:2] = data2_wrong[:2]
```

Comment:
it's not super clear to me why we don't want to allow this?
We should be able to write a tensor on a memmap-tensor and inversely

Reply:
Here we are actually writing a `MyDataMemMap2` on `MyDataMemMap1` and it is currently forbidden (hence the pytest.raises), but I am totally in favour to change this behaviour as per previous discussion.

Reply:
Yeah I would expect it to work, or at least not be disabled by design

Reply:
Should I do it? Do we want to merge this first?

Reply:
I would merge. I can have a look at that